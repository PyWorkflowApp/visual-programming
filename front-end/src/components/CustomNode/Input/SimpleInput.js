import React, { useState, useEffect } from 'react';
import { Form } from 'react-bootstrap';

export default function SimpleInput(props) {

    const [value, setValue] = useState(props.value);
    const handleChange = (event) => {
        setValue(event.target.value);
    };

    const {keyName, onChange, type} = props;
    // whenever value changes, fire callback to update config form
    useEffect(() => {
            const formValue = type === "number" ? Number(value) : value;
            onChange(keyName, formValue);
        },
        [value, keyName, onChange, type]);

    const extraProps = props.type === "textarea"
        ? {as: "textarea", rows: props.rows || 7}
        : {type: props.type};
    return (
        <Form.Control {...extraProps} name={props.keyName}
                      disabled={props.disabled}
                      defaultValue={props.value}
                      onChange={handleChange} />
    )
}
