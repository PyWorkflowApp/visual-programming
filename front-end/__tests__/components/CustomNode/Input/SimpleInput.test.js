import React from 'react'
import { shallow, mount } from 'enzyme';
import SimpleInput from '../../../../src/components/CustomNode/Input/SimpleInput';

describe('Validates SimpleInput', () => {
  it('Display SimpleInput', () => {
    const input = shallow(<SimpleInput />);
    expect(input).toMatchSnapshot();
  });

  it('Display SimpleInput as number', () => {
    const changeFn = jest.fn(() => []);
    const input = mount(<SimpleInput
          type="number"
          keyName="numberInput"
          onChange={changeFn}
          disabled={false}
          />);

    expect(changeFn.mock.calls.length).toBe(1);
    input.find({keyName: "numberInput"}).simulate('change', {target: {value: 654321}})
    expect(changeFn.mock.calls.length).toBe(2);
  });
});
