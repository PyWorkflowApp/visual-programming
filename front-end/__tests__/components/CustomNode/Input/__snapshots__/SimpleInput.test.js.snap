// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validates SimpleInput Display SimpleInput 1`] = `ShallowWrapper {}`;
